### PR TITLE
safepath: improving error messages

### PR DESCRIPTION
**Pull request by alicefr**
Giving more details which path caused the error can improve the
debugging of the root cause.

Signed-off-by: Alice Frosi <afrosi@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This could help in debugging safepath related issue like for example: https://github.com/kubevirt/kubevirt/issues/8387

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/safepath/safepath.go [hunk 1]
```diff
@@ -386,12 +387,16 @@ func UnlinkAtNoFollow(path *Path) error {
...
+	options := 0
 	if info.IsDir() {
 		// if dir is empty we can delete it with AT_REMOVEDIR
-		return unlinkat(fd.fd, basename, unix.AT_REMOVEDIR)
-	} else {
-		return unlinkat(fd.fd, basename, 0)
+		options = unix.AT_REMOVEDIR
 	}
+	if err = unlinkat(fd.fd, basename, options); err != nil {
+		return fmt.Errorf("failed unlinking path %s: %v", path.String(), err)
```
Comment (rmohr) [inline pkg/safepath/safepath.go:397]: There should be no need to explicitly call `String` with `%v` or `%q` in all the places.

**Review by rmohr — COMMENTED** [review]
Makes a lot of sense to me. One comment.

File: pkg/safepath/safepath.go (hunk 1 above)
Comment (alicefr) [reply pkg/safepath/safepath.go]: Updated, thanks for the review

File: pkg/safepath/safepath.go [hunk 2]
```diff
@@ -108,7 +109,7 @@ func newLimitedFifo(maxOps uint) *fifo {
 func OpenAtNoFollow(path *Path) (file *File, err error) {
 	fd, err := open(path.rootBase)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed opening path %v: %w", path, err)
```
[the remark this answers is not in the document]
Comment (alicefr) [reply pkg/safepath/safepath.go:112]: I used `%w` in order to wrap the error, so we can unwrap it later. For example with `errors.Is()`

[the remark this answers is not in the document]
Comment (rmohr) [reply pkg/safepath/safepath.go:112]: Yes, realized it as soon as I pressed send 😁

[the remark this answers is not in the document]
Comment (alicefr) [reply pkg/safepath/safepath.go:112]: Github is a bit slow for me today, I missed your second comment :). All good then

**Pull request merged**